sycl/include/sycl/stl_wrappers/__sycl_cmath_wrapper_impl.hpp

```diff
@@ -29,27 +27,29 @@
 // When multiple floating point arguments are available passing arguments with
 // different precision should promote to the larger type. The template helpers
 // below provide the machinery to define these promoting overloads.
-template <typename T,
-          bool = (std::is_integral_v<T> || std::is_floating_point_v<T>)>
-struct __sycl_promote {
-private:
-  // Integer types are promoted to double.
-  template <typename U>
-  static std::enable_if_t<std::is_integral_v<U>, double> test();
-
-  // Floating point types are used as-is.
-  template <typename U>
-  static std::enable_if_t<std::is_floating_point_v<U>, U> test();
-
-public:
-  // We rely on dummy templated methods and decltype to select the right type
-  // based on the input T.
-  typedef decltype(test<T>()) type;
+template <typename T, bool = std::is_integral_v<T>> struct __sycl_promote {
+  using type = double;
 };
 
 // Variant without ::type to allow SFINAE for non-promotable types.
 template <typename T> struct __sycl_promote<T, false> {};
 
+// float and double are left as is
+template <> struct __sycl_promote<float> {
+  using type = float;
+};
+template <> struct __sycl_promote<double> {
+  using type = double;
+};
+// long double is not supported yet, SFINAE it away explicitly.
+// We don't provide these overloads to makes sure that
+// mixed precision calls that include long double are
+// resolved to the "host" overload (defined by the real <cmath>),
+// matching the behaviour without promotion.
+// our long double overloads would fail to compile because
+// we'd be trying to call non SPIR-V built-ins that don't support long double.
+template <> struct __sycl_promote<long double> {};
```

Comment:
Would it not fall under the "default" case in line 35?

Reply:
Done in the latest commit.